controllers/front/GuestTrackingController.php

```diff
@@ -23,125 +23,133 @@
+
+            if (strlen($password) < Validate::PASSWORD_LENGTH) {
+                $this->errors[] = $this->trans(
+                    'Your password must be at least %min% characters long.',
+                    array('%min%' => Validate::PASSWORD_LENGTH),
+                    'Shop.Form.Help'
+                );
+            } elseif ($customer->transformToCustomer($this->context->language->id, $password)) {
+                $this->success[] = $this->trans(
+                    'Your guest account has been successfully transformed into a customer account. You can now log in as a registered shopper.',
+                    array(),
+                    'Shop.Notifications.Success'
+                );
             } else {
-                $this->assignOrderTracking($this->order_collection);
-                if (Tools::isSubmit('submitTransformGuestToCustomer')) {
-                    $customer = new Customer((int)$this->order_collection->getFirst()->id_customer);
-                    if (!Validate::isLoadedObject($customer)) {
-                        $this->errors[] = $this->trans('Invalid customer', array(), 'Shop.Notifications.Error');
-                    } elseif (!Tools::getValue('password')) {
-                        $this->errors[] = $this->trans('Invalid password.', array(), 'Shop.Notifications.Error');
-                    } elseif (!$customer->transformToCustomer($this->context->language->id, Tools::getValue('password'))) {
-                        $this->errors[] = $this->trans('An error occurred while transforming a guest into a registered customer.', array(), 'Shop.Notifications.Error');
-                    } else {
-                        $this->success[] = $this->trans('Your guest account has been successfully transformed into a customer account. You can now log in as a registered shopper.', array(), 'Shop.Notifications.Success');
-                    }
-                }
+                $this->success[] = $this->trans(
+                    'An unexpected error occurred while creating your account.',
+                    array(),
+                    'Shop.Notifications.Error'
+                );
             }
         }
     }
 
     /**
-     * Assign template vars related to page content
+     * Assign template vars related to page content.
+     *
      * @see FrontController::initContent()
      */
     public function initContent()
     {
-        FrontController::initContent();
+        parent::initContent();
 
-        /* Handle brute force attacks */
-        if (count($this->errors)) {
-            sleep(1);
+        if (!Validate::isLoadedObject($this->order)) {
+            return $this->setTemplate('customer/guest-login.tpl');
         }
 
-        if (count($this->order_collection) > 0) {
-            $this->setTemplate('customer/guest-tracking.tpl');
-        } else {
-            $this->setTemplate('customer/guest-login.tpl');
+        if ((int) $this->order->isReturnable()) {
```

Comment:
Yeah it's either false or the number of day left for the return
https://github.com/julienbourdeau/PrestaShop/blob/0bf5e8706c800171060819a16d23227a7efe1a32/classes/order/Order.php#L1161-L1161